pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -429,18 +462,39 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 				Eventually(k.UpdateStatus(mapiMachine, func() {
 					mapiMachine.Status.AuthoritativeAPI = mapiv1beta1.MachineAuthorityMachineAPI
 				})).Should(Succeed())
+
+				capiMachine = capiMachineBuilder.WithName(mapiMachine.Name).Build()
 			})
 
 			It("should successfully create the CAPI machine", func() {
-				Eventually(k.Get(
-					clusterv1resourcebuilder.Machine().WithName(mapiMachine.Name).WithNamespace(capiNamespace.Name).Build(),
-				)).Should(Succeed())
+				Eventually(k.Get(capiMachine)).Should(Succeed())
 			})
 
-			It("should successfully create the CAPA machine", func() {
-				Eventually(k.Get(
-					awsv1resourcebuilder.AWSMachine().WithName(mapiMachine.Name).WithNamespace(capiNamespace.Name).Build(),
-				)).Should(Succeed())
+			It("should successfully create the CAPA machine with correct owner references", func() {
+				// Get the CAPI machine, so we know the UID in the api server
+				Eventually(k.Get(capiMachine)).Should(Succeed())
+
+				capaMachine = capaMachineBuilder.WithOwnerReferences([]metav1.OwnerReference{
+					{
+						Kind:               machineKind,
+						APIVersion:         clusterv1.GroupVersion.String(),
+						Name:               capiMachine.Name,
+						UID:                capiMachine.UID,
+						BlockOwnerDeletion: ptr.To(true),
+						Controller:         ptr.To(false),
+					},
+				}).WithName(mapiMachine.Name).Build()
```

Comment:
I don't understand the purpose of this, it's like you're trying to write the owner ref, but then we fetch the object next and that will overwrite everything in `capaMachine` so this has no effect?